solutions/gold/izho-17-LongestBeautifulSequence.mdx

```diff
@@ -0,0 +1,270 @@
+---
+id: izho-17-LongestBeautifulSequence
+source: IZhO
+title: Longest Beautiful Subsequence
+author: Daniel Zhu
+---
+
+## Explanation
+
+### Subtasks 1-2
+
+Due to the small bounds on $N$, an $\mathcal{O}(N^2)$ solution suffices.
+
+Define $\texttt{dp}[i]$ as the length of the longest LBS (longest beautiful subsequence) that ends at $i$, and $\texttt{bc}[x][y]$ as $\text{bit\_count}(x \hspace{3pt} \& \hspace{3pt} y)$. We transition by iterating over all $j < i$ where $\texttt{bc}[a_i][a_j] == \texttt{k}[i]$.
```

Comment:
abstract definitions like this are sus in general
i try to do something like "yeah remember the sample case? this is how we do that according to this algorithm"